src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -232,65 +266,122 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::map(
   PRECICE_ASSERT(getDimensions() == output()->getDimensions(),
                  getDimensions(), output()->getDimensions());
 
-  Eigen::VectorXd &inValues  = input()->data(inputDataID)->values();
-  Eigen::VectorXd &outValues = output()->data(outputDataID)->values();
+  std::vector<double> inValues;
+  inValues.resize(input()->data(inputDataID)->values().size());
+  std::vector<double> outValues;
+  outValues.resize(output()->data(outputDataID)->values().size());
+
+  // Map the data from Eigen::VectorXd to std::vector for send and receive operations
+  Eigen::Map<Eigen::VectorXd>(&inValues[0], input()->data(inputDataID)->values().size())     = input()->data(inputDataID)->values();
+  Eigen::Map<Eigen::VectorXd>(&outValues[0], output()->data(outputDataID)->values().size())  = output()->data(outputDataID)->values();
```

Comment:
You could directly construct the vector using the `data()` and `size()` accessors of `Eigen::VectorXd`.
Sadly Eigen does not provide `begin()` and `end()` consistently, so one has to pass pointers instead.

```cpp 
  auto & localInData  = input()->data(inputDataID)->values();
  auto & localOutData = output()->data(outputDataID)->values();
  std::vector<double> inValues(localInData.data(),   localInData.data()  + localInData.size());
  std::vector<double> outValues(localOutData.data(), localOutData.data() + localOutData.size());
```